docs/developer/registry-access.md

```diff
@@ -0,0 +1,101 @@
+# Working with the registry via EPP
+
+## Overview of parts
+
+**EPP** is the protocol which describes how a registry and registrar communicate with XML over a TCP socket connection.
+
+**epplib** is a Python library implementation of the TCP socket connection. It has helper functions and dataclasses which can be used to send and receive the XML messages.
+
+**epplibwrapper** is a module in this repository which abstracts away the details of authenticating with the registry. It assists with error handling by providing error code constants and an error class with some helper methods.
+
+**Domain** is a Python class. It inherits from `django.db.models.Model` and is therefore part of Django's ORM and has a corresponding table in the local registrar database. Its purpose is to provide a developer-friendly interface to the registry based on *what a registrant or analyst wants to do*, not on the technical details of EPP.
+
+## Debugging in a Python shell
+
+You'll first need access to a Django shell in an environment with valid registry credentials. For example:
+
+```shell
+cf ssh getgov-ENVIRONMENT
+/tmp/lifecycle/shell  # this configures your environment
+./manage.py shell
+```
+
+You'll next need to import some code.
+
+```
+from epplibwrapper import CLIENT as registry, commands
+from epplib.models import common
+```
+
+Finally, you'll need to craft a request and send it.
+
+```
+request = ...
+response = registry.send(request)
```

Comment:
Glad you asked! I'm going to mention this in the docs, but not include it in the code sample because I do not want to proliferate copypasta. You ran into the failure, had to go look up why it was failing, and now understand, which is this check working as intended. 🎉